src/traces/scattermapbox/convert.js

```diff
@@ -32,19 +32,39 @@ module.exports = function convert(gd, calcTrace) {
     var hasLines = subTypes.hasLines(trace);
     var hasMarkers = subTypes.hasMarkers(trace);
     var hasText = subTypes.hasText(trace);
-    var hasCircles = (hasMarkers && trace.marker.symbol === 'circle');
-    var hasSymbols = (hasMarkers && trace.marker.symbol !== 'circle');
+    var hasCircles = hasMarkers && trace.marker.symbol === 'circle';
+    var hasSymbols = hasMarkers && trace.marker.symbol !== 'circle';
+    var hasCluster = trace.cluster.enabled;
 
-    var fill = initContainer();
-    var line = initContainer();
-    var circle = initContainer();
-    var symbol = initContainer();
+    var fill = initContainer('fill');
+    var line = initContainer('line');
+    var circle = initContainer('circle');
+    var symbol = initContainer('symbol');
 
     var opts = {
         fill: fill,
         line: line,
         circle: circle,
-        symbol: symbol
+        symbol: symbol,
+        cluster: {
+            type: 'circle',
+            filter: ['has', 'point_count'],
+            layout: {visibility: 'visible'},
+            paint: {
+                'circle-color': createClusterColors(trace.cluster.color, trace.cluster.steps),
+                'circle-radius': createClusterSizes(trace.cluster.size, trace.cluster.steps)
+            },
+        },
+        clusterCount: {
+            type: 'symbol',
+            filter: ['has', 'point_count'],
+            paint: {},
+            layout: {
+                'text-field': '{point_count_abbreviated}',
+                'text-font': ['DIN Offc Pro Medium', 'Arial Unicode MS Bold'],
```

Comment:
```suggestion
                'text-font': ['Open Sans Regular', 'Arial Unicode MS Regular'], // reuse dflt found in mapbox/layout_attributes.js 
```
It might be better to use similar defaults found here: 
https://github.com/plotly/plotly.js/blob/86f5e2d02d56a94fcf2c26b30d3539fd54211c40/src/plots/mapbox/layout_attributes.js#L27